opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -458,6 +459,11 @@ public int getPerPartesCount() {
         return MAX_CHANGESETS;
     }
 
+    private boolean isRepositoryEmpty() {
+        File headsFile = Paths.get(getDirectoryName(), Constants.DOT_GIT, "refs", "heads").toFile();
+        return headsFile.isDirectory() && (Objects.requireNonNull(headsFile.listFiles()).length == 0);
```

Comment:
for better robustness I should also list the repository directory and only consider this empty repository if that listing also comes out empty.

Reply:
except it should do the listing via something like `git ls-files` to avoid untracked files.

Reply:
The setup required for `ls-files` would require `TreeWalk` with a commit which needs the empty repository check.